### DISCUSSION TITLE
Consultation on solutions to ：the server selected protocol version TLS10 is not accepted by client preferences [TLS13, TLS12]

### DISCUSSION THREAD

**Discussion by shan1110** [dbeaver/dbeaver#22047] [Q&A] [answered]
**Is your feature request related to a problem? Please describe.**
Hello developers, I would like to ask a question about: the server selected protocol version TLS10 is not accepted by client preferences. [TLS13, TLS12] The solution to this problem.

Regarding my above problems, I find that most of the claims are to change java.security, so I am very curious about how you solve this problem, I am looking for solutions through source code or lssues.

But this is a little difficult for me as a developer who has been in the development industry for one or two years. I look forward to getting your answers, thank you

**Describe the solution you'd like**
how is dbeaver solved

**Describe alternatives you've considered**
change java.security file

**Additional context**

**Comment by Destrolaric** [comment, marked as answer]
@shan1110 DBeaver adds TLS10 protocol in included Java `java.security` file

**Reply by shan1110** [reply]
Ok, thank you. After consulting you, I also used this method to solve this problem, and your answer successfully proved that my approach was feasible

**Reply by shan1110** [reply]
But I think there may be some security risks involved in this?  After all, JDK disabled it because it has loophole .  However, I did not find any other good solutions, and Navicat and Data Grip did not address this issue
